### PR TITLE
useless Rc<Rc<T>>, Rc<Box<T>>, Rc<&T>, Box<&T>

### PR DESCRIPTION
**Pull request by jpospychala**
refers to  #2394

changelog: Add lints for Rc<Rc<T>> and Rc<Box<T>> and Rc<&T>, Box<&T>

this is based on top of another change #5310 so probably should go after that one.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1433,6 +1433,9 @@ Released 2018-09-13
 [`range_plus_one`]: https://rust-lang.github.io/rust-clippy/master/index.html#range_plus_one
 [`range_step_by_zero`]: https://rust-lang.github.io/rust-clippy/master/index.html#range_step_by_zero
 [`range_zip_with_len`]: https://rust-lang.github.io/rust-clippy/master/index.html#range_zip_with_len
+[`rc_borrows`]: https://rust-lang.github.io/rust-clippy/master/index.html#rc_borrows
```
Comment (Manishearth) [inline CHANGELOG.md]: do you think it's worth merging these lints into a single "redundant_allocation" lint?

File: clippy_lints/src/types.rs [hunk 2]
```diff
@@ -322,6 +416,37 @@ impl Types {
...
                             return; // don't recurse into the type
                         }
+                    } else if Some(def_id) == cx.tcx.lang_items().rc() {
+                        if match_type_parameter(cx, qpath, &paths::RC) {
+                            span_lint_and_help(
+                                cx,
+                                RC_RC,
+                                hir_ty.span,
+                                "usage of `Rc<Rc<T>>`",
+                                "Rc<Rc<T>> can be simplified to Rc<T>",
```
Comment (Manishearth) [inline clippy_lints/src/types.rs]: we should use suggestions here as well, as mentioned in the other PR

**Comment by Manishearth** [comment]
Yeah, we should make this a single lint

File: CHANGELOG.md (hunk 1 above)
Comment (jpospychala) [reply CHANGELOG.md]: sure, will do

**Comment by Manishearth** [comment]
Should we just close the other PR and land these all together?

(Also this needs a rebase)

**Comment by Manishearth** [comment]
@bors r+

thanks!

**Pull request merged**